ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -831,7 +831,7 @@ public CompletableFuture<RaftClientReply> submitClientRequestAsync(
       if (type.is(TypeCase.READ)) {
         // TODO: We might not be the leader anymore by the time this completes.
         // See the RAFT paper section 8 (last part)
-        replyFuture = processQueryFuture(stateMachine.query(request.getMessage()), request);
+        replyFuture = readOnlyAsync(request);
```

Comment:
This linearizable read feature must be configurable.  Some applications may not want to use it.

Reply:
I'll add a ReadOption, which has 3 enum values:
1. default: use the current read path.
2. safe: use read index.
3. lease: use leader lease (this will be implemented later).